### PR TITLE
test: add E2E testcase for #872 #873

### LINKED ISSUE #872: bug: manager-api should not allow create route with remote_addrs and remote_addr at the same time
# Bug report

## Describe the bug

manager-api should not allow create route with remote_addrs and remote_addr at the same time

## How to Reproduce

```
{
			caseDesc: "config route with remote_addr and remote_addrs at the same time",
			Object:   MangerApiExpect(t),
			Method:   http.MethodPut,
			Path:     "/apisix/admin/routes/r1",
			Body: `{
					"uri": "/test_uri",
					"remote_addrs": ["127.0.0.1","192.168.0.1"],
					"remote_addr": "127.0.0.1",
					"upstream": {
						"type": "roundrobin",
						"nodes": {
							"172.16.238.20:1980": 1
						}
					}
				}`,
			Headers:      map[string]string{"Authorization": token},
			ExpectStatus: http.StatusBadRequest,
		},
```
It is return 200 now.

### LINKED ISSUE #873: bug: create route with error format remote_addrs manager-api will return 200, but should return 400
# Bug report

## Describe the bug

manager-api shuold not allow create route with error format remote_addrs

## How to Reproduce

```
{
			caseDesc: "config route with invalid remote_addrs",
			Object:   MangerApiExpect(t),
			Method:   http.MethodPut,
			Path:     "/apisix/admin/routes/r1",
			Body: `{
					"uri": "/test_uri",
					"remote_addrs": "127.0.0.1",
					"upstream": {
						"type": "roundrobin",
						"nodes": {
							"172.16.238.20:1980": 1
						}
					}
				}`,
			Headers:      map[string]string{"Authorization": token},
			ExpectStatus: http.StatusBadRequest,
},
```
It is return 200 now.

### PR DESCRIPTION
**Pull request by idbeta**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] E2E Testcase
- [ ] New feature provided
- [ ] Improve performance

- Related issues
resolve #872  
resolve #873
___
### Bugfix
- Description

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

### CODE REVIEW

**Comment by juzhiyuan** [comment]
Kindly add resolve keywords before issue number is needed.

File: api/test/e2e/route_e2e_remote_addr_rarescen_test.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+ import (
+	 "net/http"
+	 "testing"
+	 //  "time"
+	 //  "github.com/stretchr/testify/assert"
+	 //  "github.com/tidwall/gjson"
+ )
+ 
+ func TestRoute_add_with_invalid_remote_addr_rarescen(t *testing.T) {
```
Comment (moonming) [inline api/test/e2e/route_e2e_remote_addr_rarescen_test.go:27]: what is “ratescen”？

File: api/test/e2e/route_e2e_remote_addr_rarescen_test.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+						"remote_addrs": "127.0.0.1",
+						"upstream": {
+							"type": "roundrobin",
+							"nodes": {
+								"172.16.238.20:1980": 1
+							}
+						}
+					}`,
+			 Headers:      map[string]string{"Authorization": token},
+			 ExpectStatus: http.StatusBadRequest,
```
Comment (moonming) [inline api/test/e2e/route_e2e_remote_addr_rarescen_test.go:45]: still not test data plane. is this unit test or e2e？

File: api/test/e2e/route_e2e_remote_addr_rarescen_test.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+ }
+ 
+ //Teardown
+ func TestConsumer_teardown(t *testing.T) {
+	 _ = []HttpTestCase{
+		 {
+			 caseDesc:     "delete route",
+			 Object:       MangerApiExpect(t),
+			 Method:       http.MethodDelete,
+			 Path:         "/apisix/admin/routes/r1",
```
Comment (moonming) [inline api/test/e2e/route_e2e_remote_addr_rarescen_test.go:80]: this router is not created, why delete will return 200?

**Review by moonming — CHANGES_REQUESTED** [review]

**Comment by membphis** [comment]
we should use a useful title

**Comment by membphis** [comment]
related issues: #872  #873 

one PR for one thing, so I think we have to close this PR

**Comment by membphis** [comment]
#872 : milestone 2.2
#873 : milestone 2.1.1

@idbeta you can add the test cases in the issue, it is a better way

**Pull request closed without merging**